functions.php

```diff
@@ -3,18 +3,24 @@
 // On Pi's, this placeholder gets replaced with ${ALLSKY_CONFIG}.
 // On other machines it won't and references to it will silently fail.
 define('ALLSKY_CONFIG',  'XX_ALLSKY_CONFIG_XX');
+// The issue is how to determine if we're on a Pi without using
+// the exec() function which is often disabled on remote machines.
+// And we can't do @exec() to see if it works because that can
+// display a message in the user's browser window.
+// To avoid that message, assume if we're not on a Pi that exec() doesn't work.
+// TODO: make this a user setting if exec() works???
+// If you can think of a better way than to check for a hard-coded
+// path, please update the code.
 
 // If on a Pi, check that the placholder was replaced.
-exec("grep -q 'Model.*: Raspberry' /proc/cpuinfo", $none, $return);
-// Split the placeholder so it doesn't get replaced if the update script is run multiple times.
-// Note: return code 0 == a match, return code 1 == no match
-if ($return==0 && ALLSKY_CONFIG == "XX_ALLSKY_CONFIG" . "_XX") {
+$exec_works = file_exists('/var/www/html/allsky/config.js');		// YUCK!
```

Comment:
or:
```
function exec_enabled() {
    $disabled = explode(',', ini_get('disable_functions'));
    return !in_array('exec', $disabled);
}
```

Reply:
For Pi detection, it may be better to use something like:
```
$isarm = preg_match("/(arm|aarch)/",php_uname());
```

Reply:
@linuxkidd Thanks for the suggestions.  I think the "function_exists('exec')" produces an error message in the browser window on some remote hosts that have exec() disabled, but I'll have the user test the first two methods.

Reply:
> For Pi detection, it may be better to use something like:
> 
> ```
> $isarm = preg_match("/(arm|aarch)/",php_uname());
> ```

@EricClaeys  I just tested this and it returns false on my webhost. I think this is a safer Pi detection than testing for exec().

Reply:
Changed to use preg_match()